diff --git a/miniWeather.cpp b/miniWeather.cpp
index 437973c..9c9eb4c 100644
--- a/miniWeather.cpp
+++ b/miniWeather.cpp
@@ -104,7 +104,7 @@ int main(int argc, char **argv) {
   //So, you'll want to have nx_glob be twice as large as nz_glob
   nx_glob = 40;      //Number of total cells in the x-direction
   nz_glob = 20;      //Number of total cells in the z-direction
-  sim_time = 10;     //How many seconds to run the simulation
+  sim_time = 10000;     //How many seconds to run the simulation
 
   ///////////////////////////////////////////////////////////////////////////////////////
   // END USER-CONFIGURABLE PARAMETERS
