diff --git a/miniWeather.cpp b/miniWeather.cpp
index 5c40d6a..5418891 100644
--- a/miniWeather.cpp
+++ b/miniWeather.cpp
@@ -349,7 +349,7 @@ void set_halo_values_z( double *state ) {
   const double mnt_width = xlen/8;
   double       x, xloc, mnt_deriv;
 
-#pragma acc parallel loop private(x, xloc, mnt_deriv) default(present)
+#pragma acc parallel loop collapse(2) private(x, xloc, mnt_deriv) default(present)
   for (ll=0; ll<NUM_VARS; ll++) {
     for (i=0; i<nx+2*hs; i++) {
       if (ll == ID_WMOM) {
