diff --git a/c/miniWeather.cpp b/c/miniWeather.cpp
index ef9808d..3ea40d7 100644
--- a/c/miniWeather.cpp
+++ b/c/miniWeather.cpp
@@ -335,7 +335,7 @@ void set_halo_values_z( double *state ) {
   const double mnt_width = xlen/8;
   double       x, xloc, mnt_deriv;
 
-#pragma acc parallel loop private(x, xloc, mnt_deriv) default(present)
+#pragma acc parallel loop collapse(2) private(x, xloc, mnt_deriv) default(present)
   for (ll=0; ll<NUM_VARS; ll++) {
     for (i=0; i<nx+2*hs; i++) {
       if (ll == ID_WMOM) {
