diff --git a/c/miniWeather.cpp b/c/miniWeather.cpp
index ef9808d..81dd04e 100644
--- a/c/miniWeather.cpp
+++ b/c/miniWeather.cpp
@@ -218,7 +218,7 @@ void compute_tendencies_x( double *state , double *flux , double *tend ) {
           stencil[s] = state[inds];
         }
         //Fourth-order-accurate interpolation of the state
-        vals[ll] = -stencil[0]/12 + 7*stencil[1]/12 + 7*stencil[2]/12 - stencil[3]/12;
+        vals[ll] = -(stencil[0] + 7*stencil[1] + 7*stencil[2] - stencil[3])/12;
         //First-order-accurate interpolation of the third spatial derivative of the state (for artificial viscosity)
         d3_vals[ll] = -stencil[0] + 3*stencil[1] - 3*stencil[2] + stencil[3];
       }
@@ -273,7 +273,7 @@ void compute_tendencies_z( double *state , double *flux , double *tend ) {
           stencil[s] = state[inds];
         }
         //Fourth-order-accurate interpolation of the state
-        vals[ll] = -stencil[0]/12 + 7*stencil[1]/12 + 7*stencil[2]/12 - stencil[3]/12;
+        vals[ll] = -(stencil[0] + 7*stencil[1] + 7*stencil[2] - stencil[3])/12;
         //First-order-accurate interpolation of the third spatial derivative of the state
         d3_vals[ll] = -stencil[0] + 3*stencil[1] - 3*stencil[2] + stencil[3];
       }
